src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -316,10 +316,15 @@ namespace Microsoft.FSharp.Core
 
     [<AttributeUsage(AttributeTargets.Method ||| AttributeTargets.Property, AllowMultiple=false)>]
     [<Sealed>]
-    type NoDynamicInvocationAttribute() =
+    type NoDynamicInvocationAttribute(isLegacy: bool) =
+
         inherit System.Attribute()
 
-    [<AttributeUsage(AttributeTargets.Parameter, AllowMultiple=false)>]
+        new () = NoDynamicInvocationAttribute(false)
+
+        member x.IsLegacy = isLegacy
```

Comment:
From our conversation, we think we may not need "IsLegacy". When set to "true", it would tell the compiler to actually generate the implementation of the method but keep the signature with the attreibute; but, we might be able to remove "[<NoDynamicInvocation>]" instead of having to add "IsLegacy".

Reply:
See below